pandas/tests/typing/valid/test_frame.py

```diff
@@ -0,0 +1,834 @@
+    # float]", variable has type "DataFrame")
+    df3: pd.DataFrame = df.median(axis=1)  # type: ignore[assignment]
+    # error: Incompatible types in assignment (expression has type "Union[Series,
+    # float]", variable has type "DataFrame")
+    df4: pd.DataFrame = df.median(1, True)  # type: ignore[assignment]
+    # error: Incompatible types in assignment (expression has type "Union[Series,
+    # float]", variable has type "Series")
+    s3: pd.Series = df.median(  # type: ignore[assignment]
+        axis=1, skipna=True, numeric_only=False
+    )
+
+
+def test_types_itertuples() -> None:
+    df = pd.DataFrame(data={"col1": [2, 1], "col2": [3, 4]})
+    res1: Iterable[Tuple[Any, ...]] = df.itertuples()
+    res2: Iterable[Tuple[Any, ...]] = df.itertuples(index=False, name="Foobar")
+    res3: Iterable[Tuple[Any, ...]] = df.itertuples(index=False, name=None)
+
+
+def test_types_sum() -> None:
+    df = pd.DataFrame(data={"col1": [2, 1], "col2": [3, 4]})
+    df.sum()
+    df.sum(axis=1)
+
+
+def test_types_cumsum() -> None:
+    df = pd.DataFrame(data={"col1": [2, 1], "col2": [3, 4]})
+    df.cumsum()
+    df.sum(axis=0)
+
+
+def test_types_min() -> None:
+    df = pd.DataFrame(data={"col1": [2, 1], "col2": [3, 4]})
+    df.min()
+    df.min(axis=0)
+
+
+def test_types_max() -> None:
+    df = pd.DataFrame(data={"col1": [2, 1], "col2": [3, 4]})
+    df.max()
+    df.max(axis=0)
+
+
+def test_types_quantile() -> None:
+    df = pd.DataFrame(data={"col1": [2, 1], "col2": [3, 4]})
+    df.quantile([0.25, 0.5])
+    df.quantile(0.75)
+    df.quantile()
+
+
+def test_types_clip() -> None:
+    df = pd.DataFrame(data={"col1": [20, 12], "col2": [3, 14]})
+    df.clip(lower=5, upper=15)
+
+
+def test_types_abs() -> None:
+    df = pd.DataFrame(data={"col1": [-5, 1], "col2": [3, -14]})
+    df.abs()
+
+
+def test_types_var() -> None:
+    df = pd.DataFrame(data={"col1": [2, 1], "col2": [1, 4]})
+    df.var()
+    df.var(axis=1, ddof=1)
+    df.var(skipna=True, numeric_only=False)
+
+
+def test_types_std() -> None:
+    df = pd.DataFrame(data={"col1": [2, 1], "col2": [1, 4]})
+    df.std()
+    df.std(axis=1, ddof=1)
+    df.std(skipna=True, numeric_only=False)
+
+
+def test_types_idxmin() -> None:
+    df = pd.DataFrame(data={"col1": [2, 1], "col2": [3, 4]})
+    df.idxmin()
+    df.idxmin(axis=0)
+
+
+def test_types_idxmax() -> None:
+    df = pd.DataFrame(data={"col1": [2, 1], "col2": [3, 4]})
+    df.idxmax()
+    df.idxmax(axis=0)
+
+
+# This was added in 1.1.0 https://pandas.pydata.org/docs/whatsnew/v1.1.0.html
+def test_types_value_counts() -> None:
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [1, 4]})
+    df.value_counts()
+
+
+def test_types_unique() -> None:
+    # This is really more for of a Series test
+    df = pd.DataFrame(data={"col1": [1, 2], "col2": [1, 4]})
+    df["col1"].unique()
+
+
+def test_types_apply() -> None:
+    df = pd.DataFrame(data={"col1": [2, 1], "col2": [3, 4]})
+    df.apply(lambda x: x ** 2)
```

Comment:
Pandas switch recently to black 22.1.0 which changes "x ** 2" to "x**2". This happens twice in this file and one time in test_series.py